aws-redshift-cluster/src/main/java/software/amazon/redshift/cluster/ReadHandler.java

```diff
@@ -166,26 +168,18 @@ private GetResourcePolicyResponse getNamespaceResourcePolicy(
             return GetResourcePolicyResponse.builder().resourcePolicy(resourcePolicy).build();
         } catch (InvalidPolicyException | UnsupportedOperationException e) {
             throw new CfnInvalidRequestException(ResourceModel.TYPE_NAME, e);
-        } catch (SdkClientException | RedshiftException e) {
+        } catch (RedshiftException e) {
+            if (e.awsErrorDetails().errorCode().equals(GET_RESOURCE_POLICY_ERROR_CODE) &&
+                    e.awsErrorDetails().errorMessage().contains(GET_RESOURCE_POLICY_ERROR)) {
+                logger.log(String.format("RedshiftException: User is not authorized to perform: redshift:GetResourcePolicy on resource %s",
+                        e.getMessage()));
+            } else {
+                throw new CfnGeneralServiceException(e);
```

Comment:
for new users who opt in to use resource policy, if their role doesn't have permission, how do we inform them?

Reply:
discussed offline, looks good, can you add detailed comments about why this is okay for the new customers who want to create resource policy?

Reply:
Done, added comments with new commits.